force-app/main/default/classes/StringGenerator.cls

```diff
@@ -0,0 +1,60 @@
+//Create a class that takes a set of strings as input and allows the caller to obtain: 
+
+//1) a random string (completely randomly chosen from the list passed as input to the class) 
+
+//2) a uniquely random string (one string should not repeat until the cycle is complete). 
+
+//Unique randomness means that once you serve a string, that string should not again be served until a all strings have been requested.  
+
+public with sharing class StringGenerator 
+{
+    public static List<Integer> listOfServedIndexes = new List<Integer>();
+
+    public static String randomString( List<String> stringList )
+    {
+        Contract.requires( stringList != null, 'randomString called with null stringList' );
+
+        Integer randomIndex;
+        
+        if( stringList.size() != 0 )
+        {
+            randomIndex = Integer.valueof( ( Math.random() * stringList.size() ) );
+
```

Comment:
Math.random() * stringList.size() can return a number equal to stringList.size(), for e.g if list size is 4, it can return 4 as well while the valid indices will only be from 0 to 3 for that list. So using 4 as index would cause a failure?

Reply:
Math.random() returns a value from 0 to the given range - 1.. for example if range is 10, it is going to return a number from 0 to 9. Taking your example if size is 4 it will never return 4, it will only return values from 0 to 3 i.e : 4-1